test/unit-tests/cve/cveTest.js

```diff
@@ -244,4 +244,188 @@ describe('Testing the /cve/:id endpoint in Cve Controller', () => {
         })
     })
   })
+
+  context('Updating CVE Endpoint', () => {
+    it('Update CVE record when requestor is secretariat', (done) => {
+      class OrgRepo {
+        async getOrgUUID () {
+          return null
+        }
+      }
+      class CveRepo {
+        async updateByCveId (cveId, newCve) {
+          return { n: 1 }
+        }
+      }
+      class UserRepo {
+        async getUserUUID () {
+          return null
+        }
+      }
+      server.route('/cve-update-record/:id')
+        .put((req, res, next) => {
+          const factory = {
+            getCveRepository: () => { return new CveRepo() },
+            getOrgRepository: () => { return new OrgRepo() },
+            getUserRepository: () => { return new UserRepo() }
+          }
+          req.ctx.repositories = factory
+          next()
+        }, cveParams.parsePostParams, cveController.CVE_UPDATE_SINGLE)
+
+      chai.request(server)
+        .put(`/cve-update-record/${cveIdPublic4}`)
+        .set(cveFixtures.secretariatHeader)
+        .send(cvePublicPass4)
+        .end((err, res) => {
+          if (err) {
+            done(err)
+          }
+
+          expect(res).to.have.status(200)
+          expect(res).to.have.property('body').and.to.be.a('object')
+          expect(res.body).to.have.property('updated').and.to.be.a('object')
+          expect(res.body.updated).to.have.nested.property('CVE_data_meta.ID').and.to.equal(cveIdPublic4)
+          expect(res.body.updated).to.have.nested.property('CVE_data_meta.STATE').and.to.equal(CONSTANTS.CVE_STATES.PUBLIC)
+          done()
+        })
+    })
+
```

Comment:
A test should be added to respond with a bad request when a requester wants to update a CVE record to a RESERVED state. I included this additional scenario in issue #330.

Reply:
Probably should be a forbidden response.